src/snowflake/cli/api/metrics.py

```diff
@@ -109,10 +109,8 @@ class for holding metrics span data and encapsulating related operations
     children: Set[CLIMetricsSpan] = field(init=False, default_factory=set)
 
     # private state
-    # start time of the step from the monotonic clock in order to calculate execution time
-    _monotonic_start: float = field(
-        init=False, default_factory=lambda: time.monotonic()
-    )
+    # start time of the step from a performance counter in order to calculate execution time
+    _start_time: float = field(init=False, default_factory=time.perf_counter)
```

Comment:
Ran into failing integ tests where the ordering of the spans was not deterministic, did some more research and found that `time.perf_counter` should just be a better version of `time.monotonic` with appropriate granularity on Windows, so replaced those calls here

Reply:
I don't get that part - Why is time determining the order? The most accurate way is first call goes first. Or are we dealing with multithreads  here?

Reply:
On Windows, `time.monotonic()` does not have as much precision, so multiple spans would end up having the same start time, so when we sort them by start time (to make it so that the first call goes first) the order ends up not being the same as the order as when we start the spans. This is fixed with `time.perf_counter()`

Reply:
weird that the sorting function wasn't stable though.

Reply:
When we take spans breadth-first (to make sure they conform to limits) we sort by depth first, which breaks the ordering of when the spans get inserted: https://github.com/snowflakedb/snowflake-cli/blob/main/src/snowflake/cli/api/metrics.py#L293

Reply:
Not sure I follow. This should all be deterministic, even if two spans have the same start time. IMHO that's a hard requirement. I've been burnt by stuff like this in the past, so I'm extra careful now. Happy to discuss possible solutions if you want.

Reply:
Took a stab at it, let me know if this works for you: https://github.com/snowflakedb/snowflake-cli/pull/1844/commits/ae43f25d8835785a4887339f688dfa953f86ca22